### DISCUSSION TITLE
This page has expired. Would you like to refresh the page?

### DISCUSSION THREAD

**Discussion by Henriquex25** [livewire/livewire#4959] [7. Bugs (archived)] [answered]
I would like to apologize for my english

I just installed livewire. When a component needs to be updated, the message "This page has expired. Would you like to refresh the page?" appears.

can anybody help me?

Laravel 9.5.1

![image](https://user-images.githubusercontent.com/88984677/168205663-fafbb5eb-37c5-4604-a79b-6e17a32b5f4f.png)

![image](https://user-images.githubusercontent.com/88984677/168205618-d3ca6334-39f3-4615-a801-7e8bf13b3461.png)

![image](https://user-images.githubusercontent.com/88984677/168205750-8cede327-3ede-40cf-af41-b26891501045.png)

**Comment by pubsajib** [comment]
I had the same problem & adding
`<meta name="csrf_token" value="{{ csrf_token() }}"/>`
at the head resolved it

**Reply by Narven** [reply]
```
<meta name="csrf-token" content="{{ csrf_token() }}" />
```

**Comment by arukompas** [comment]
@Henriquex25 - are you using the `web` middleware on your routes? It's required to start a session. Without a session, every new Livewire update thinks it's a new session and thus expired.

**Comment by GKuetenang** [comment]
I have the same error in my Livewire file upload process since 2 weeks. Any solution please ??

**Reply by joshhanley** [reply]
@GustaveTsopmo hard to say without seeing your code. Did any of the above suggestions help?

**Comment by Virus5600** [comment]
This is for reference only, especially to those who are applying security to their web applications. I wish y'all don't spend 3 hours just to fix it like me...

I'm implementing secure practices and I forgot that I've set the `SESSION_SECURE_COOKIE` to true in the `config/session.php` and at the `.env` file, forcing the session to only be passed over HTTPS and never through HTTP.

I recommend checking your config session and .env files if this happen...

**Reply by mikerockett** [reply]
Bumped into this as well, after tightening things up. For me, it wasn't the secure cookie property, but rather the fact that I had a period (.) in my session cookie name.

**Comment by BAKKALIAYOUB** [comment]
i solve it by adding this to `VerifyCsrfToken`Middleware:

```
 protected $except = [
         'livewire/*',
     ];
```

**Reply by salma31nidar** [reply]
it works for me thanks

**Reply by joshhanley** [reply]
@BAKKALIAYOUB and @salma31nidar I would be careful doing that, your disabling csrf protection for Livewire requests, which is a bad idea as could cause security issues.

I would be adding the csrf meta tag into your layout file as described in a comment above instead of disabling it all together.

Hope this helps!

**Comment by cosmefar** [comment]
the problem is: "barryvdh/laravel-debugbar"
change in .env APP_DEBUG=true to APP_DEBUG=false
In my case it was!

**Reply by keverson-ms** [reply]
thank you, solve my question

**Comment by Gmk0** [comment]
IS A SOLUTION EXISTING FOR THIS??

**Reply by Virus5600** [reply]
There are multiple above depending on your case. Have you gone through each of them?

**Reply by Gmk0** [reply]
I solved the problem

**Reply by robsontenorio** [reply]
@Gmk0 how ??

**Reply by Gmk0** [reply]
Mine was the sessions table that had a small problem I had declared the Id to use as uuid. I also had to do it in the session table

**Comment by BoGnY** [comment]
this is a bug.. i've reported at #6941.
is it caused by caching csrf token... it should be fixed by 5d887316f2aaf83c0e380ac5e72766f19700fa3b but bug persists...

**Comment by lortschi** [comment]
In my case I had in `Kernel.php` the 2 lines:
> protected $middlewareGroups = [
> 'web' => [
>  \Illuminate\Session\Middleware\StartSession::class,
> \Illuminate\View\Middleware\ShareErrorsFromSession::class,

inside the `protected $middleware = [ ... ]`

**Comment by gladx** [comment]
for laravel version 11 :

bootstrap/app.php 
<strike>
```
    ->withMiddleware(function (Middleware $middleware) {
        $middleware->validateCsrfTokens(except: [
            'livewire/*',
        ]);
    })
```
</strike>

use this solution: [
lrljoe](https://github.com/livewire/livewire/discussions/4959#discussioncomment-9972558)

**Reply by ritta-ibrahim** [reply]
this solved in my case, thanks

**Reply by duke-d3v** [reply]
It solve my problem too, thanks!
I wonder why it has dislike.

**Reply by lrljoe** [reply, marked as answer]
Because you **want** to validate CSRF tokens for Livewire...  Please don't do the above, it is a security risk!

You need to ensure that this:
```
<meta name="csrf_token" value="{{ csrf_token() }}"/>
```
Is in your layout head section!

**Reply by duke-d3v** [reply]
Thanks man i'll consider that

**Reply by gladx** [reply]
@lrljoe I've updated my comment, thanks for your notice 👍

**Reply by TheArKaID** [reply]
As per [livewire codebase](https://github.com/livewire/livewire/blob/main/js/utils.js#L186), it should be:
`<meta name="csrf-token" content="{{ csrf_token() }}"/>`

**Reply by BoGnY** [reply]
> `<meta name="csrf-token" content="{{ csrf_token() }}"/>`

this is the correct code

**Reply by Zephni** [reply]
This has just started randomly happening for me. I have the <meta name="csrf-token" content="{{ csrf_token() }}"/> snippet added already. Any idea's if anything else could be causing this?

**Comment by AMIRA-SHOP** [comment]
<meta name="csrf_token" value="{{ csrf_token() }}"/>

**Comment by AMIRA-SHOP** [comment]
<meta name="csrf_token" value="{{ csrf_token() }}"/>

**Comment by WillianDamasceno** [comment]
My problem was that I asked the server admins to activate the CDN for me, and the CDN had a rule not to respect any of my application's request headers, so it cached everything.
Once I asked them to deactivate this behavior, the login started to work again.
Deactivating this behavior means that the CDN now only caches routes depending on the headers I send from the server.
Maybe this can help someone that activated a CDN recently.

**Comment by Badro0** [comment]
In my case, it was a wrong SESSION_DOMAIN in .env file. Using the right domain fixed the issue.

**Reply by LDH90** [reply]
same case for me

**Comment by rajentrivedi** [comment]
This issue happens due to csrf token mismatch.
simple work around : 

                        `<script>
                               Livewire.hook("request", ({ fail }) => {
                                fail(async ({ status, preventDefault, retry }) => {
                                    if (status === 419) {
                                        preventDefault();
                                        
                                        const response = await fetch("/refresh-csrf", {
                                            method: "GET",
                                            headers: {
                                                "Accept": "application/json",
                                            },
                                            credentials: "same-origin",
                                        });

                                        const data = await response.json();
                                        const token = data.token;

                                        document.querySelector("meta[name=\\"csrf-token\\"]")
                                            .setAttribute("content", token);

                                        Livewire.csrfToken = token;
                                        retry();
                                    }
                                });
                            });
                            
                        });
                        </script>`
                        
Add Route to web.php
                        
                        `Route::get('/refresh-csrf', function () {
                                     return response()->json(['token' => csrf_token()]);
                         })->name('refresh-csrf');`

**Reply by ercanertan** [reply]
**Small correction**

document.querySelector('meta[name="csrf-token"]').setAttribute("content", token);

**Reply by robinmoisson** [reply]
@rajentrivedi I don't see a `retry` method on the `fail` callback of the `request` hook, unfortunately. Did you make it work in another way?

See [the doc](https://livewire.laravel.com/docs/3.x/javascript#request-hooks) over here.
